src/main/kotlin/net/leanix/vsm/sbomBooster/VsmSbomBoosterApplication.kt

```diff
@@ -17,6 +19,7 @@ import java.util.concurrent.atomic.AtomicInteger
 @EnableConfigurationProperties(PropertiesConfiguration::class)
 class VsmSbomBoosterApplication(
     private val gitHubApiService: GitHubApiService,
+    private val gitLabApiService: GitLabApiService,
```

Comment:
Do you have an example of this? This is actually my first time using Spring and Kotlin (not just combined, I've never used either of them prior to this). I want to attempt this edit, just not sure what it looks like.

Reply:
there are multiple ways to achieve this, and the closest I could find in our projects is this:
https://github.com/leanix/vsm-github-broker/blob/532f9bdc0aa088998d5012f05ef2e58d06d4acaf/src/main/kotlin/net/leanix/vsm/githubbroker/connector/runner/InitialStateRunner.kt#L19

But a simpler way (in mu opinion) is to just create an interface like:
interface GitApiService {
    fun getUsername(token: String): String?
    fun getRepositories(token: String, organizationOrGroup: String): List<Repository>
}

make both classes implement the interface.

create a configuration to create the correct bean based on the env variable:
@Configuration
class GitApiServiceConfiguration {
    @Bean
    @ConditionalOnProperty(name = ["GIT_PROVIDER"], havingValue = "GITHUB")
    fun gitHubApiService(): GitApiService {
        return GitHubApiService()
    }

    @Bean
    @ConditionalOnProperty(name = ["GIT_PROVIDER"], havingValue = "GITLAB")
    fun gitLabApiService(): GitApiService {
        return GitLabApiService()
    }
}

This is really not something necessary at this moment, it was just a suggestion to keep in mind for the future :)

I just didn't merge the PR because I still haven't tested with a Gitlab instance, otherwise LGTM

Reply:
I can't quite get this to work without spending more time on it than I'd like to. I'll mark this as resolved and it can be saved for a future enhancement.